### PR TITLE
[11.0][MIG] sale_invoice_group_method

### PR DESCRIPTION
**Pull request by rgarnau**
Migration from 10.0. No major changes

Sale Invoice Group Method
=========================

This module allows you to combine several Sales Orders into a single invoice,
if they meet the split criteria defined by the 'Invoice Group Method'.

The split criteria is defined in the 'Invoice Group Method' by a combination
of fields of the Sales Order. 'Invoice Address', 'Currency' and 'Payment Term'
are always included.

You can assign a default 'Invoice Group Method' to customers, so that it will
be proposed on their orders.

When no Invoice Group Method is defined in a Sales Order, the standard
approach will be used, which groups by 'Invoice Address' and 'Currency'.

Note: Existing draft invoices are not considered in the process of grouping.
However, you can find the feature implemented in ``sale_merge_draft_invoice``
from sale-workflow repository.

Configuration
=============

#. Go to 'Sales / Configuration / Invoice Group Method'
#. Create an Invoice Group Method and choose the fields of the Sales Order
   that should be equal in that particular grouping method, for their orders
   to be merged into the same invoice. For example, create the Invoice Group
   Method 'By Customer' and select the field 'Customer').

Usage
=====

Update customers
----------------

#. Go to 'Sales / Sales / Customers'
#. Choose a customer and go to 'Sales & Purchases' page.
#. Update the 'Default Invoice Group Method'

Create a Quote / Sale Order
---------------------------

#. Go to 'Sales / Sales / Quotations'.
#. Create a new Quote and inside the 'Other Information' page select an
   option from the field 'Invoice Group Method'. If the customer had a
   default, it will have been provided already.
#. Complete the Quote as usual.

Create Invoices
---------------

#. Go to 'Sales / Sales / Sales Orders'.
#. Select all the sales orders with status 'To invoice'.
#. Go to 'Action' and click 'Invoice Order'. As a result, draft invoices will be
   created for the selected Sales Orders, consolidating them according to the
   Invoice Group Method defined.

### CODE REVIEW

File: sale_invoice_group_method/README.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
+.. image:: https://img.shields.io/badge/licence-LGPL--3-blue.svg
+   :target: https://www.gnu.org/licenses/lgpl-3.0-standalone.html
+   :alt: License LGPL-3
+
+=========================
+Sale Invoice Group Method
+=========================
+
+This module allows you to combine several Sales Orders into a single invoice,
+if they meet the split criteria defined by the 'Invoice Group Method'.
```
Comment (gdgellatly) [inline sale_invoice_group_method/README.rst]: One minor point - why does it refer to split criteria?  Surely its merge or group criteria - yes I know its not part of the port but I noticed it.

File: sale_invoice_group_method/README.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+The split criteria is defined in the 'Invoice Group Method' by a combination
```
Comment (gdgellatly) [inline sale_invoice_group_method/README.rst]: As above

File: sale_invoice_group_method/view/sale_invoice_group_method_view.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    <record model="ir.actions.act_window" id="sale_invoice_group_method_action">
+        <field name="name">Sale Invoice Group Method</field>
+        <field name="type">ir.actions.act_window</field>
+        <field name="res_model">sale.invoice.group.method</field>
+        <field name="view_type">form</field>
+        <field name="view_mode">tree,form</field>
+        <field name="context">{'sale_invoice_group_method':1}</field>
+    </record>
+
+</odoo>
```
Comment (gdgellatly) [inline sale_invoice_group_method/view/sale_invoice_group_method_view.xml]: Complaining of no newline, not sure if it is standard for xml but other files have it.

File: sale_invoice_group_method/view/res_partner_view.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        <field name="inherit_id" ref="base.view_partner_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//notebook/page[@name='sales_purchases']//group[@name='sale']//field[@name='message_bounce']"
+                   position="after">
+                <field name="invoice_group_method_id"/>
+            </xpath>
+        </field>
+    </record>
+
+</odoo>
```
Comment (gdgellatly) [inline sale_invoice_group_method/view/res_partner_view.xml]: Also complaining of no newline.

File: sale_invoice_group_method/README.rst [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+are always included.
+
+You can assign a default 'Invoice Group Method' to customers, so that it will
+be proposed on their orders.
+
+When no Invoice Group Method is defined in a Sales Order, the standard
+approach will be used, which groups by 'Invoice Address' and 'Currency'.
+
+Note: Existing draft invoices are not considered in the process of grouping.
+However, you can find the feature implemented in ``sale_merge_draft_invoice``
```
Comment (gdgellatly) [inline sale_invoice_group_method/README.rst:23]: Hmmm, recommending a module that isn't available for the version? I get it, just saying, is meant to be user documentation afterall.

**Review by gdgellatly — APPROVED** [review]
The port is all fine, (very) minor comments related to pre existing documentation and newlines

File: sale_invoice_group_method/README.rst (hunk 5 above)
Comment (rgarnau) [reply sale_invoice_group_method/README.rst:23]: It is available here: https://github.com/OCA/sale-workflow/pull/569. It's just that it's waiting for more reviews in order to be merged :)

**Comment by rgarnau** [comment]
@gdgellatly changes applied! :+1:

File: sale_invoice_group_method/README.rst (hunk 5 above)
Comment (gdgellatly) [reply sale_invoice_group_method/README.rst:23]: Other reviewers, please ignore above, it is absolutely non blocking.

**Review by etobella — APPROVED** [review]
LGTM :+1:

**Review by JordiBForgeFlow — APPROVED** [review]

**Pull request merged**